src/timer.rs

```diff
@@ -334,24 +329,33 @@ where
 static mut TIMER_DRIVER_AVAILABLE: bool = true;
 
 #[derive(Copy, Clone, Default, PartialEq, Eq)]
-pub struct ActiveTimer {
+struct ActiveTimer {
     instant: u32,
     set_at: u32,
 }
 
+/// Context for the time driver.
+/// You can create a context as follows:
+/// ```
+/// let context = DriverContext::create();
+/// ```
 pub struct DriverContext {
```

Comment:
What is the purpose of DriverContext?

Reply:
It holds the current active timer (which may vary) and the current time (to avoid unnecessary context switches). Unfortunately, you cannot just put it into `TimerDriver` as you have a borrow on the current time in the callback.

Reply:
It holds the current time and it colds is current active timer. It is the context shared by all active timer futures.

Reply:
On a higher level, it's because sharing references without using the heap requires an additional state object (in this case: the `DriverContext`). I am currently testing a solution which enables using the state object as a callback, s.t. only one layer of indirection is needed and the `activate` part can be skipped.